test/utilities/extensionUtils.ts

```diff
@@ -291,3 +344,98 @@ export async function findVSCodeBinary(): Promise<string> {
 
   return codeBin[0];
 }
+
+export async function verifyExtensionsAreRunning(
```

Comment:
This setups the UI, etc. to verify that extensions are running as expected.

Its job is to wait until all needed extensions are activated or it times out.

It sets up two promise functions, one to check extensions and the other a timeout. Then using Promise.race, it will wait until one of the promises finishes first.

It returns a boolean indicating results